node/actors/bft/src/leader/replica_commit.rs

```diff
@@ -127,25 +126,42 @@ impl StateMachine {
             .or_insert_with(|| CommitQC::new(message.clone(), self.config.genesis()))
             .add(&signed_message, self.config.genesis());
 
-        // We store the message in our cache.
         let cache_entry = self
             .commit_message_cache
             .entry(message.view.number)
             .or_default();
+
+        // We check validators weight from current messages, stored by proposal
+        let weight_before = self.config.genesis().validators.weight_from_msgs(
+            cache_entry
+                .clone()
+                .iter()
+                .map(|(_, m)| m)
+                .filter(|m| m.msg.proposal == message.proposal)
+                .collect::<Vec<_>>()
+                .as_slice(),
+        );
+
```

Comment:
nit: If this calculation is only used for the later debug_assert, then I don't think it's worth it. Just remove the assert.